### ISSUE TITLE
Lint/ShadowingOuterLocalVariable: a false positive case with case statement and block variable

### ISSUE THREAD

**Issue by rickselby** [rubocop/rubocop#10512]
## Expected behavior

No offense with `Lint/ShadowingOuterLocalVariable` cop

## Actual behavior

```ruby
 W: Lint/ShadowingOuterLocalVariable: Shadowing outer local variable - item.
  array.each do |item|
                 ^^^^
```

## Steps to reproduce the problem

```ruby
def test(array)
  case array.length
  when 1
    item = array.first
    p item
  else
    array.each do |item|
      p item
    end
  end
end
```

## RuboCop version

```
1.27.0 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 2.7.5 x86_64-linux)
  - rubocop-performance 1.13.3
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.9.0
```

I thought https://github.com/rubocop/rubocop/issues/10479 would fix this one, but it didn't.

**Comment by ydah** [comment]
I've confirmed that it does reproduce. Sorry. I'll attempt to fix it.

**Comment by ydah** [comment]
Oh, I see. I may have made a big mistake.
As has been mentioned at https://github.com/rubocop/rubocop/issues/3067
> Ruby hoists local variables to the scope above the if statement.

@koic Do you think I need to revert https://github.com/rubocop/rubocop/pull/10484?
